scripts/env-doctor.sh

```diff
@@ -11,111 +11,111 @@ NC='\033[0m' # No Color
+  fi
+  echo
 }
 
 # Diagnostics and remedies
 run_diagnostic \
-    "Check if fnm is installed" \
-    "command -v fnm >/dev/null 2>&1" \
-    "Install fnm" \
-    "brew install fnm"
+  "Check if fnm is installed" \
+  "command -v fnm >/dev/null 2>&1" \
+  "Install fnm" \
+  "brew install fnm"
 
 run_diagnostic \
-    "Check if Node.js is installed" \
-    "command -v node >/dev/null 2>&1" \
-    "Install Node.js using fnm" \
-    "fnm install --lts && fnm use lts-latest"
+  "Check if Node.js is installed" \
+  "command -v node >/dev/null 2>&1" \
+  "Install Node.js using fnm" \
+  "fnm install --lts && fnm use lts-latest"
 
 run_diagnostic \
-    "Check if Node.js version matches .node-version" \
-    "[ \"$(node --version)\" = \"v$(cat .node-version)\" ]" \
-    "Install the correct Node.js version using fnm" \
-    "fnm install $(cat .node-version) && fnm use $(cat .node-version)"
+  "Check if Node.js version matches .node-version" \
+  "[ \"$(node --version)\" = \"v$(cat .node-version)\" ]" \
+  "Install the correct Node.js version using fnm" \
+  "fnm install $(cat .node-version) && fnm use $(cat .node-version)"
 
 run_diagnostic \
-    "Check if Yarn version matches package.json" \
-    "[ \"$(yarn --version)\" = \"$(node -p "require('./package.json').packageManager.split('@')[1]")\" ]" \
-    "Install correct Yarn version" \
-    "corepack enable && yarn set version $(node -p "require('./package.json').packageManager.split('@')[1]")"
+  "Check if Yarn version matches package.json" \
+  "[ \"$(yarn --version)\" = \"$(node -p "require('./package.json').packageManager.split('@')[1]")\" ]" \
+  "Install correct Yarn version" \
+  "corepack enable && yarn set version $(node -p "require('./package.json').packageManager.split('@')[1]")"
 
 run_diagnostic \
-    "Check if Git is installed" \
-    "command -v git >/dev/null 2>&1" \
-    "Install Git" \
-    "brew install git"
+  "Check if Git is installed" \
+  "command -v git >/dev/null 2>&1" \
+  "Install Git" \
+  "brew install git"
 
 run_diagnostic \
-    "Check if VSCode is installed" \
-    "command -v code >/dev/null 2>&1" \
-    "Install Visual Studio Code" \
-    "brew install --cask visual-studio-code"
+  "Check if VSCode is installed" \
+  "command -v code >/dev/null 2>&1" \
+  "Install Visual Studio Code" \
+  "brew install --cask visual-studio-code"
 
 run_diagnostic \
-    "Check if Docker is installed" \
-    "command -v docker >/dev/null 2>&1" \
-    "Install Docker" \
-    "brew install --cask docker"
+  "Check if Docker is installed" \
+  "command -v docker >/dev/null 2>&1" \
+  "Install Docker" \
+  "brew install --cask docker"
 
 run_diagnostic \
-    "Build repo" \
-    "yarn build" \
-    "Clean, reinstall dependencies, and rebuild" \
-    "git clean -fdx && yarn install && yarn build"
+  "Build repo" \
+  "yarn build" \
+  "Clean, reinstall dependencies, and rebuild" \
+  "git clean -fdx "**/node_modules" "**/bundles" && yarn install && yarn build"
```

Comment:
I didn't know glob patterns were supported by git!

Reply:
learned this from @kriskowal

Reply:
Apparently globs are not working for me. Where can I find info on how it's supposed to work?

Reply:
Ok I found that globs may not be enabled by default. Also I think that glob pattern doesn't actually work. What worked for me (more aggressive version with more things cleaned):

`-- ':(glob)**/node_modules/**' ':(glob)**/bundles/**' ':(glob)**/dist/**' ':(glob)**/build/**'`

Reply:
Also I tend to do `bin/agd build` these days instead as that also builds the golang stuff for those of use who care. However it has its own "download deps" logic which may not be desirable here.

Reply:
We could also try using [exclusions](https://git-scm.com/docs/git-clean#Documentation/git-clean.txt--eltpatterngt) instead of inclusions. But I'll leave this as an exercise for whoever runs into the problem next. This is meant to be a crowdsourced solution set

Reply:
Inclusions is probably safer. It's just that I don't understand how this line even works given the quoting, and that glob patterns are not enabled by default.